components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -643,18 +623,33 @@ void RRTMGPRadiation::run_impl (const double dt) {
           t_lev(i+1,nlay+1) = d_tint(i,nlay);
 
           // Note that RRTMGP expects ordering (col,lay,bnd) but the FM keeps things in (col,bnd,lay) order
-          Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nswbands*nlay), [&] (const int&idx) {
-              auto b = idx / nlay;
-              auto k = idx % nlay;
-              aero_tau_sw(i+1,k+1,b+1) = d_aero_tau_sw(icol,b,k);
-              aero_ssa_sw(i+1,k+1,b+1) = d_aero_ssa_sw(icol,b,k);
-              aero_g_sw  (i+1,k+1,b+1) = d_aero_g_sw  (icol,b,k);
-          });
-          Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nlwbands*nlay), [&] (const int&idx) {
-              auto b = idx / nlay;
-              auto k = idx % nlay;
-              aero_tau_lw(i+1,k+1,b+1) = d_aero_tau_lw(icol,b,k);
-          });
+          if (m_do_aerosol_rad) {
```

Comment:
You can't access a class member from inside a kernel [^1]. You need to create a local copy before the kernel dispatch, and use that in the kernel.

[^1] Soon we will switch to C++17, and we will be able to use `KOKKOS_CLASS_LAMBDA`, which will allow to capture members of the class. For now, we're still stuck with this tedious workaround.

Reply:
Also, It appears some markdown syntax doesn't work in PR comments...

Reply:
Doh, I knew that. Pushing a fix.

Reply:
Okay should be fixed.